### DISCUSSION TITLE
Compiling a user space application that uses libbpfgo as dependency fails

### DISCUSSION THREAD

**Discussion by alexjch** [aquasecurity/libbpfgo#395] [Q&A] [answered]
libbpfgo version v0.6.0-libbpf-1.3
libbpf 1.3 (it's installed in the ./root directory

```
make build-go
CC=gcc CGO_CFLAGS="-I ./root/usr/include" CGO_LDFLAGS="./root/usr/lib64/libbpf.a" go build -mod vendor -o mce_log cmd/mce_log/main.go
# github.com/aquasecurity/libbpfgo
vendor/github.com/aquasecurity/libbpfgo/map-common.go:154:10: could not determine kind of name for C.bpf_map_get_info_by_fd
make: *** [Makefile:14: build-go] Error 1```

Commenting map-common.go:152 - 158 allows me to avoid the issue. It seems that  bpf_map_get_info_by_fd does not exists in libbpf?

**Comment by rafaeldtinoco** [comment, marked as answer]
Use this directory as an example how to achieve what you want please: https://github.com/rafaeldtinoco/random/tree/main/bpf-example.

**Reply by alexjch** [reply]
Thanks, I will test it out and report back.

**Comment by alexjch** [comment]
Alright, it works. I will take a better look to see what's wrong with what I was doing, if there's anything to report I will add to the discussion. Otherwise I would say I'm unblocked. Thanks again.

**Reply by rafaeldtinoco** [reply]
Sure, keep in mind this software is open source and distributed as-is, and all help you will get here is based on 'best-effort'. With that said, I'm sure you will get something when you need. Good  luck.
